qiskit/aqua/operators/primitive_ops/pauli_op.py

```diff
@@ -70,10 +71,21 @@ def add(self, other: OperatorBase) -> OperatorBase:
         if isinstance(other, PauliOp) and self.primitive == other.primitive:
             return PauliOp(self.primitive, coeff=self.coeff + other.coeff)
 
+        if (
+                isinstance(other, PauliOp)
+                and isinstance(self.coeff, (int, float, complex))
+                and hasattr(other, 'coeff')
```

Comment:
Since we no other is a PauliOp and all ops in opflow have coeff, the hasattr test seems unneccessary